### PR TITLE
linksharing: add to local docker-compose

### PR DESCRIPTION
**Pull request by Erikvv**
Add linksharing to docker-compose for testing locally.

Testing: run `docker-compose up --detach`.
Check the logs using `docker-compose logs --follow`
Visit http://localhost:8000

It lacks some good usage instructions but I'm not writing them now.

Change-Id: Ie1a193ee84ddb55e0431eaed48302150f28ce0dc

### CODE REVIEW

**Comment by stefanbenten** [comment]
@clabot check

File: docker-local/start-linksharing.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xe
+
+cd /tmp
+wget --output-document=GeoLite2-City.mmdb \
+    https://link.storjshare.io/s/jxy4d5ux5wvc2jf7bmt2rmlnfeeq/oliosite/public/GeoLite2-City.mmdb?download
```
Comment (stefanbenten) [inline docker-local/start-linksharing.sh:5]: This absolutely can not be commited to main and actually is illegal to do..
Please remove asap.

File: docker-local/docker-compose.yml [hunk 2]
```diff
@@ -59,6 +60,16 @@ services:
...
+  linksharing:
+    image: golang:1.17
+    entrypoint: /scripts/start-linksharing.sh
+    volumes:
+      - localgopath:/go/storjpath
+      - ${GATEWAY_SRC_DIR}:/code
+      - ./start-linksharing.sh:/scripts/start-linksharing.sh
+      - storjsimconfig:/config
+    ports:
+      - "8001:8001"
```
Comment (stefanbenten) [inline docker-local/docker-compose.yml:72]: It should probably also depends on storj-sim, authservice being up.

**Review by stefanbenten — CHANGES_REQUESTED** [review]

File: docker-local/start-linksharing.sh (hunk 1 above)
Comment (Erikvv) [reply docker-local/start-linksharing.sh]: Sorry I was not aware.

I will check if it works without.

File: docker-local/docker-compose.yml [hunk 3]
```diff
@@ -59,6 +60,16 @@ services:
...
       - authservice
       - tracing
+  linksharing:
+    image: golang:1.17
+    entrypoint: /scripts/start-linksharing.sh
+    volumes:
+      - localgopath:/go/storjpath
+      - ${GATEWAY_SRC_DIR}:/code
+      - ./start-linksharing.sh:/scripts/start-linksharing.sh
+      - storjsimconfig:/config
```
Comment (stefanbenten) [inline docker-local/docker-compose.yml:70]: We probably should include a ENV to reference a geoip DB from the local machine here.

Comment (Erikvv) [reply docker-local/docker-compose.yml:70]: It appears it works without. 

Maybe we can mock a file which puts the Docker subnet (172.*) in Atlanta.

**Review by halkyon — DISMISSED** [review]

**Comment by Erikvv** [comment]
Rebased. I would appreciate if this can get a final review or alternatively inform me that this change should not be merged.

File: docker-local/docker-compose.yml [hunk 4]
```diff
@@ -59,6 +59,18 @@ services:
...
       - tracing
+  linksharing:
+    image: golang:1.17.7
+    entrypoint: /scripts/start-linksharing.sh
+    volumes:
+      - localgopath:/go/storjpath
+      - ${GATEWAY_SRC_DIR}:/code
+      - ./start-linksharing.sh:/scripts/start-linksharing.sh
+    ports:
+      - "8001:8001"
```
Comment (halkyon) [inline docker-local/docker-compose.yml:70]: should this be 20020?

Comment (Erikvv) [reply docker-local/docker-compose.yml]: Yes thanks

**Review by halkyon — DISMISSED** [review]

File: pkg/auth/peer.go [hunk 5]
```diff
@@ -110,6 +110,10 @@ func New(ctx context.Context, log *zap.Logger, config Config, configDir string)
 		return nil, errs.New("allowed satellites parameter '--allowed-satellites' resolved to zero satellites")
 	}
 
+	for satelliteID := range allowedSats {
+		log.Debug("allowed satellite: " + satelliteID.String())
+	}
```
Comment (amwolff) [inline pkg/auth/peer.go:115]: nit: it might be a good idea to make this a separate change.

File: docker-local/docker-compose.yml [hunk 6]
```diff
@@ -13,7 +13,7 @@ services:
       - POSTGRES_HOST_AUTH_METHOD=trust
       - POSTGRES_DB=dcstorj
   storjsim:
-    image: golang:1.16.0
+    image: golang:1.17
```
Comment (amwolff) [inline docker-local/docker-compose.yml:16]:
```suggestion
    image: golang:1.17.7
```

File: pkg/auth/peer.go (hunk 5 above)
Comment (Erikvv) [reply pkg/auth/peer.go:115]: Do you want this as a separate change?

File: docker-local/docker-compose.yml (hunk 6 above)
Comment (Erikvv) [reply docker-local/docker-compose.yml]: Done

File: pkg/auth/peer.go (hunk 5 above)
Comment (amwolff) [reply pkg/auth/peer.go:115]: No; it's fine

**Review by amwolff — APPROVED** [review]

**Review by halkyon — APPROVED** [review]

**Pull request merged**